src/ui/osx/TogglDesktop/LoginViewController.m

```diff
@@ -678,9 +774,21 @@ - (void)clickLoginButton:(id)sender
     [self setUserSignUp:NO];
     [self showLoaderView:YES];
 
-    if (!toggl_login_async(ctx, [email UTF8String], [pass UTF8String]))
-    {
-        return;
+    if (self.isLoginSignUpAsSSO) {
+        if ([email isEqualToString:self.emailSSOTextField.stringValue]) {
+            // user is logging in with same email that he've used for SSO login
+            [[DesktopLibraryBridge shared] showMessageAfterLogin:NSLocalizedString(@"SSO login successfully enabled for your account.",
```

Comment:
Can this `showMessageAfterLogin` thing be just a one-time subscription to the next `toggl_on_time_entry_list` call? 
So if `toggl_on_time_entry_list` was an observable, that would be something like `toggl_on_time_entry_list.take(1).subscribe(showInfoMessage)` instead of this back-and-forth between platform and library + introducing state into library.